Directory.Build.targets

```diff
@@ -23,6 +23,7 @@
 
         <!--build related-->
         <PackageReference Include="MinVer" Version="4.3.0" PrivateAssets="All"/>
+		<PackageReference Update="Backport.System.Threading.Lock" Version="2.0.3"/>
```

Comment:
I'd prefer not to introduce a new dependency. Unless there was a practical performance problem that this new locking construct fixed, I don't think backporting this to our .net 8 builds is really necessary. On the other hand, conditional compilation is also a pain. Maybe this change should wait for a future version where we are no longer supporting .net < 9.

Reply:
The backport library is conditionally enabled only for pre .NET 9.0. So with this you're able to lock as if you're only supporting net9. In fact even if you forget to remove the PackageReference once you drop support for net8, it will not be a problem since it is conditionally eliminated for net9 or later.

Reply:
Oh, I didn't see at first that the backport is actually your open source work - seems like a great contribution to the dotnet community that you've made! 

However, we never take external dependencies except what's made by Microsoft (unavoidable since we are building for aspnet core). It just simplifies supply chain audits and compliance if we don't bring in anything else.

No offense meant at all to you! We definitely will revisit this for a future release.